.github/workflows/require-checklist.yml

```diff
@@ -0,0 +1,23 @@
+name: Require Reviewer Checklist
+
+on:
+  pull_request:
+    types: [opened, reopened, edited, synchronize]
+    branches: [master, 1.*]
+
+jobs:
+  require-reviewer-checklist:
+    runs-on: ubuntu-latest
+    steps:
+      - env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          REPO_NAME: ${{ github.event.repository.full_name }}
+          PR_NUMBER: ${{ github.event.number }}
+        run: |
+          function grep_no_match_ok() {
+            local exit_code
+            grep "$@" || exit_code=$?
+            return $((exit_code == 1 ? 0 : exit_code))
+          }
+          unchecked=$(gh api repos/"$REPO_NAME"/issues/"$PR_NUMBER" --jq '.body' | awk '/## Reviewer Checklist/,/## After-merge Checklist/' | grep_no_match_ok '\[ \]')
+          [ -n "$unchecked" ] && echo "Found unchecked reviewer items. Failing..." && echo "$unchecked" && exit 1 || :
```

Comment:
This is supposed to check PRs, but it's querying an API endpoint for issues.

Also, the check could be simplified to something like:
```bash
if gh api "repos/$REPO_NAME/pulls/$PR_NUMBER" -q .body|sed -n '/## Reviewer Checklist/,/## After-merge Checklist/{ /^- \[ \]/p }'|grep -q .; then
    echo "Found unchecked reviewer items. Failing..."
    exit 1
fi
```

Reply:
I was thinking the body of the PR would be made available to GitHub actions without having to call the GitHub API, but if that's not the case, then using `gh` does seem like the best option.

Reply:
PRs are issues in some places in GitHub.

Reply:
@mihaiplesa I know. But let's not add to the problem by using `issues` in urls when we mean `pulls`. It might make no difference to the software, but it does to a human reading the code.

Reply:
If it works with pulls then sure.

Reply:
That's another point - it's better to use `gh pr ...`, than `gh api ...`

Reply:
I've switching to using `gh pr` and sed for printing. I prefer to list the offending lines explicitly, so letting grep do that. @wknapik I think this is good now